stl/inc/span

```diff
@@ -405,13 +405,13 @@ public:
         requires (_Extent == dynamic_extent || _Extent == _Size)
     constexpr span(type_identity_t<element_type> (&_Arr)[_Size]) noexcept : _Mybase(_Size), _Mydata(_Arr) {}
 
-    template <size_t _Size>
-        requires (_Extent == dynamic_extent || _Extent == _Size)
-    constexpr span(array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+    template <class _OtherTy, size_t _Size>
+        requires (_Extent == dynamic_extent || _Extent == _Size) && is_convertible_v<_OtherTy*, element_type*>

```

Comment:
This allows initialization of `span<base>` from `array<derived, N>`, which LWG 3255 does not allow.
```suggestion
        requires (_Extent == dynamic_extent || _Extent == _Size) && is_convertible_v<_OtherTy(*)[], element_type(*)[]>
```